### PR TITLE
Animations

### PR DESCRIPTION
**Pull request by jabezzy2000**
Implemented a few animations to improve the look and feel of the app
- Created anim xmls to define the movement of the fragments as the User moves from fragment to fragment
- I also tried using an animation with the splash-screen just because it looked too abrupt upon opening
#44 

https://user-images.githubusercontent.com/90366540/178219700-9c0c9a65-c506-4881-a555-d103e6699547.mp4

### CODE REVIEW

File: app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java [hunk 1]
```diff
@@ -39,7 +40,10 @@ public class SignUpActivity extends AppCompatActivity {
     ImageButton uploadProfilePictureBtn;
     ImageView signUpProfilePicture;
     private String TAG = "SignUpActivity";
-    public ParseFile file;
+    public final static int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 1034;
+    public String photoFileName = "photo.jpg";
```
Comment (gminky019) [inline app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java:44]: should be const.

File: app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java [hunk 2]
```diff
@@ -39,7 +40,10 @@ public class SignUpActivity extends AppCompatActivity {
...
+    File photoFile;
+    private String fileName = "photo.jpg";
```
Comment (gminky019) [inline app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java:46]: const here as wel.

File: app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java [hunk 3]
```diff
@@ -59,32 +62,93 @@ public void onClick(View v) {
...
+                uploadProfilePicture();
+            }
+        });
+
 
     }
 
+    private void uploadProfilePicture() {
+        launchCamera();
+    }
```
Comment (gminky019) [inline app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java:77]: naming here is funky. Your uploading the picture and launching camera?

File: app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java [hunk 4]
```diff
@@ -59,32 +62,93 @@ public void onClick(View v) {
...
+
+    private void launchCamera() {
+        // create Intent to take a picture and return control to the calling application
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create a File reference for future access
+        photoFile = getPhotoFileUri(photoFileName);
```
Comment (gminky019) [inline app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java:83]: For this project this works but lets talk design and future impl as you move froward.

**Comment by OmarB97** [comment]
Just at a high level, the animations are a bit rough and may need a bit of tweaking (e.g. speed up a bit, I am seeing a bit overlap of content during animation which is not very easy on the eyes, etc).

File: app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java [hunk 5]
```diff
@@ -59,32 +62,93 @@ public void onClick(View v) {
...
-                    Log.e(TAG, e.getMessage());
+                    Toast.makeText(SignUpActivity.this, "" + e.getMessage(), Toast.LENGTH_SHORT).show();
                 }
             }
         });
 
     }
 
+    private void saveUserProfilePictureInBackground(ParseFile photo, User user) {
+        photo.saveInBackground(new SaveCallback() {
```
[the remark this answers is not in the document]
Comment (jabezzy2000) [reply app/src/main/java/com/example/peerrequest/activities/SignUpActivity.java:115]: Noted

File: app/src/main/java/com/example/peerrequest/activities/LoginActivity.java [hunk 6]
```diff
@@ -88,9 +89,12 @@ private void loginUser(String username, String password) {
...
             public void done(ParseUser user, ParseException e) {
                 if (e != null) {
-                    return;
+                    Utilities.showAlert("Login Invalid", "Incorrect Username/Password",LoginActivity.this);
                 }
-                goMainActivity();
+                else{
+                    goMainActivity();
+                }
+
```
[the remark this answers is not in the document]
Comment (jabezzy2000) [reply app/src/main/java/com/example/peerrequest/activities/LoginActivity.java:97]: Well, the goMainActivity is a method that handles the intent ( moving the user to the main Activity upon a successful login). However, in the case where the User inputted a wrong email/password, I use a method i defined in my Utils class to create a dialog that shows an error message telling the user they inputted a wrong username or password.

**Pull request merged**